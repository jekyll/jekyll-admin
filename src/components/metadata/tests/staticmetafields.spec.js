import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import StaticMetaData from '../StaticMetaFields';

const propsList = [
  {
    label: 'simple field-value',
    data: { layout: 'page' },
  },
  {
    label: 'array field-value',
    data: { plugins: ['jekyll-admin', 'jekyll-feed'] },
  },
  {
    label: 'array field-value with nested key-value pairs',
    data: {
      navs: [
        {
          label: 'About Us',
          url: '/about/',
        },
        {
          label: 'Contact',
          url: '/contact-us/',
        },
      ],
    },
  },
  {
    label: 'array field-value with nested arrays',
    data: {
      rows: [['foo', 'bar'], ['lorem', 'ipsum']],
    },
  },
  {
    label: 'object field-value',
    data: { author: { name: 'John Doe', handle: 'JohnDoe' } },
  },
  {
    label: 'object field-value with nested key-value pairs',
    data: {
      roles: {
        admin: {
          current: 'John Doe',
          handle: 'JohnDoe',
        },
        collaborator: {
          current: 'Jane Doe',
          handle: 'JaneDoe',
        },
      },
    },
  },
  {
    label: 'object field-value with nested arrays',
    data: {
      tableData: {
        header: ['Filename', 'Type', 'Actions'],
        rows: ['about.md', 'Markdown File', ['view', 'delete']],
      },
    },
  },
];

describe('Components::StaticMetaData', () => {
  propsList.forEach(entry => {
    it(`should render ${entry.label} correctly`, () => {
      const tree = renderer
        .create(<StaticMetaData fields={entry.data} />, {
          createNodeMock: () => document.createElement('textarea'),
        })
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
