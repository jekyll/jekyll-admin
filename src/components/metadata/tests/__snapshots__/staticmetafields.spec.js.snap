// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components::StaticMetaData should render array field-value correctly 1`] = `
<div
  className="metafields"
  id="defaults"
  title="Default Front Matter"
>
  <div
    className="metafield"
  >
    <div
      className="meta-key array"
    >
      <input
        className="field key-field"
        disabled={true}
        type="text"
        value="plugins"
      />
    </div>
    <div
      className="meta-value-array"
    >
      <div
        className="array-item-wrap"
      >
        <div
          className="array"
        >
          <div
            className="array-header"
          >
            <span
              className="array-field-num"
            >
              1
              .
            </span>
          </div>
          <div
            className="meta-value"
          >
            <textarea
              className="field value-field"
              disabled={true}
              onChange={[Function]}
              style={
                Object {
                  "height": 0,
                }
              }
              value="jekyll-admin"
            />
          </div>
        </div>
      </div>
      <div
        className="array-item-wrap"
      >
        <div
          className="array"
        >
          <div
            className="array-header"
          >
            <span
              className="array-field-num"
            >
              2
              .
            </span>
          </div>
          <div
            className="meta-value"
          >
            <textarea
              className="field value-field"
              disabled={true}
              onChange={[Function]}
              style={
                Object {
                  "height": 0,
                }
              }
              value="jekyll-feed"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components::StaticMetaData should render array field-value with nested arrays correctly 1`] = `
<div
  className="metafields"
  id="defaults"
  title="Default Front Matter"
>
  <div
    className="metafield"
  >
    <div
      className="meta-key array"
    >
      <input
        className="field key-field"
        disabled={true}
        type="text"
        value="rows"
      />
    </div>
    <div
      className="meta-value-array"
    >
      <div
        className="array-item-wrap"
      >
        <div
          className="array"
        >
          <div
            className="array-header"
          >
            <span
              className="array-field-num"
            >
              1
              .
            </span>
          </div>
          <div
            className="meta-value-array"
          >
            <div
              className="array-item-wrap"
            >
              <div
                className="array"
              >
                <div
                  className="array-header"
                >
                  <span
                    className="array-field-num"
                  >
                    1
                    .
                  </span>
                </div>
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="foo"
                  />
                </div>
              </div>
            </div>
            <div
              className="array-item-wrap"
            >
              <div
                className="array"
              >
                <div
                  className="array-header"
                >
                  <span
                    className="array-field-num"
                  >
                    2
                    .
                  </span>
                </div>
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="bar"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="array-item-wrap"
      >
        <div
          className="array"
        >
          <div
            className="array-header"
          >
            <span
              className="array-field-num"
            >
              2
              .
            </span>
          </div>
          <div
            className="meta-value-array"
          >
            <div
              className="array-item-wrap"
            >
              <div
                className="array"
              >
                <div
                  className="array-header"
                >
                  <span
                    className="array-field-num"
                  >
                    1
                    .
                  </span>
                </div>
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="lorem"
                  />
                </div>
              </div>
            </div>
            <div
              className="array-item-wrap"
            >
              <div
                className="array"
              >
                <div
                  className="array-header"
                >
                  <span
                    className="array-field-num"
                  >
                    2
                    .
                  </span>
                </div>
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="ipsum"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components::StaticMetaData should render array field-value with nested key-value pairs correctly 1`] = `
<div
  className="metafields"
  id="defaults"
  title="Default Front Matter"
>
  <div
    className="metafield"
  >
    <div
      className="meta-key array"
    >
      <input
        className="field key-field"
        disabled={true}
        type="text"
        value="navs"
      />
    </div>
    <div
      className="meta-value-array"
    >
      <div
        className="array-item-wrap"
      >
        <div
          className="array"
        >
          <div
            className="array-header"
          >
            <span
              className="array-field-num"
            >
              1
              .
            </span>
          </div>
          <div
            className="meta-value-object"
          >
            <div
              className="object-item-wrap"
            >
              <div
                className="object-key simple"
              >
                <input
                  className="field key-field"
                  disabled={true}
                  type="text"
                  value="label"
                />
              </div>
              <div
                className="object-value"
              >
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="About Us"
                  />
                </div>
              </div>
            </div>
            <div
              className="object-item-wrap"
            >
              <div
                className="object-key simple"
              >
                <input
                  className="field key-field"
                  disabled={true}
                  type="text"
                  value="url"
                />
              </div>
              <div
                className="object-value"
              >
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="/about/"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="array-item-wrap"
      >
        <div
          className="array"
        >
          <div
            className="array-header"
          >
            <span
              className="array-field-num"
            >
              2
              .
            </span>
          </div>
          <div
            className="meta-value-object"
          >
            <div
              className="object-item-wrap"
            >
              <div
                className="object-key simple"
              >
                <input
                  className="field key-field"
                  disabled={true}
                  type="text"
                  value="label"
                />
              </div>
              <div
                className="object-value"
              >
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="Contact"
                  />
                </div>
              </div>
            </div>
            <div
              className="object-item-wrap"
            >
              <div
                className="object-key simple"
              >
                <input
                  className="field key-field"
                  disabled={true}
                  type="text"
                  value="url"
                />
              </div>
              <div
                className="object-value"
              >
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="/contact-us/"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components::StaticMetaData should render object field-value correctly 1`] = `
<div
  className="metafields"
  id="defaults"
  title="Default Front Matter"
>
  <div
    className="metafield"
  >
    <div
      className="meta-key object"
    >
      <input
        className="field key-field"
        disabled={true}
        type="text"
        value="author"
      />
    </div>
    <div
      className="meta-value-object"
    >
      <div
        className="object-item-wrap"
      >
        <div
          className="object-key simple"
        >
          <input
            className="field key-field"
            disabled={true}
            type="text"
            value="name"
          />
        </div>
        <div
          className="object-value"
        >
          <div
            className="meta-value"
          >
            <textarea
              className="field value-field"
              disabled={true}
              onChange={[Function]}
              style={
                Object {
                  "height": 0,
                }
              }
              value="John Doe"
            />
          </div>
        </div>
      </div>
      <div
        className="object-item-wrap"
      >
        <div
          className="object-key simple"
        >
          <input
            className="field key-field"
            disabled={true}
            type="text"
            value="handle"
          />
        </div>
        <div
          className="object-value"
        >
          <div
            className="meta-value"
          >
            <textarea
              className="field value-field"
              disabled={true}
              onChange={[Function]}
              style={
                Object {
                  "height": 0,
                }
              }
              value="JohnDoe"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components::StaticMetaData should render object field-value with nested arrays correctly 1`] = `
<div
  className="metafields"
  id="defaults"
  title="Default Front Matter"
>
  <div
    className="metafield"
  >
    <div
      className="meta-key object"
    >
      <input
        className="field key-field"
        disabled={true}
        type="text"
        value="tableData"
      />
    </div>
    <div
      className="meta-value-object"
    >
      <div
        className="object-item-wrap"
      >
        <div
          className="object-key array"
        >
          <input
            className="field key-field"
            disabled={true}
            type="text"
            value="header"
          />
        </div>
        <div
          className="object-value"
        >
          <div
            className="meta-value-array"
          >
            <div
              className="array-item-wrap"
            >
              <div
                className="array"
              >
                <div
                  className="array-header"
                >
                  <span
                    className="array-field-num"
                  >
                    1
                    .
                  </span>
                </div>
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="Filename"
                  />
                </div>
              </div>
            </div>
            <div
              className="array-item-wrap"
            >
              <div
                className="array"
              >
                <div
                  className="array-header"
                >
                  <span
                    className="array-field-num"
                  >
                    2
                    .
                  </span>
                </div>
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="Type"
                  />
                </div>
              </div>
            </div>
            <div
              className="array-item-wrap"
            >
              <div
                className="array"
              >
                <div
                  className="array-header"
                >
                  <span
                    className="array-field-num"
                  >
                    3
                    .
                  </span>
                </div>
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="Actions"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="object-item-wrap"
      >
        <div
          className="object-key array"
        >
          <input
            className="field key-field"
            disabled={true}
            type="text"
            value="rows"
          />
        </div>
        <div
          className="object-value"
        >
          <div
            className="meta-value-array"
          >
            <div
              className="array-item-wrap"
            >
              <div
                className="array"
              >
                <div
                  className="array-header"
                >
                  <span
                    className="array-field-num"
                  >
                    1
                    .
                  </span>
                </div>
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="about.md"
                  />
                </div>
              </div>
            </div>
            <div
              className="array-item-wrap"
            >
              <div
                className="array"
              >
                <div
                  className="array-header"
                >
                  <span
                    className="array-field-num"
                  >
                    2
                    .
                  </span>
                </div>
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="Markdown File"
                  />
                </div>
              </div>
            </div>
            <div
              className="array-item-wrap"
            >
              <div
                className="array"
              >
                <div
                  className="array-header"
                >
                  <span
                    className="array-field-num"
                  >
                    3
                    .
                  </span>
                </div>
                <div
                  className="meta-value-array"
                >
                  <div
                    className="array-item-wrap"
                  >
                    <div
                      className="array"
                    >
                      <div
                        className="array-header"
                      >
                        <span
                          className="array-field-num"
                        >
                          1
                          .
                        </span>
                      </div>
                      <div
                        className="meta-value"
                      >
                        <textarea
                          className="field value-field"
                          disabled={true}
                          onChange={[Function]}
                          style={
                            Object {
                              "height": 0,
                            }
                          }
                          value="view"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="array-item-wrap"
                  >
                    <div
                      className="array"
                    >
                      <div
                        className="array-header"
                      >
                        <span
                          className="array-field-num"
                        >
                          2
                          .
                        </span>
                      </div>
                      <div
                        className="meta-value"
                      >
                        <textarea
                          className="field value-field"
                          disabled={true}
                          onChange={[Function]}
                          style={
                            Object {
                              "height": 0,
                            }
                          }
                          value="delete"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components::StaticMetaData should render object field-value with nested key-value pairs correctly 1`] = `
<div
  className="metafields"
  id="defaults"
  title="Default Front Matter"
>
  <div
    className="metafield"
  >
    <div
      className="meta-key object"
    >
      <input
        className="field key-field"
        disabled={true}
        type="text"
        value="roles"
      />
    </div>
    <div
      className="meta-value-object"
    >
      <div
        className="object-item-wrap"
      >
        <div
          className="object-key object"
        >
          <input
            className="field key-field"
            disabled={true}
            type="text"
            value="admin"
          />
        </div>
        <div
          className="object-value"
        >
          <div
            className="meta-value-object"
          >
            <div
              className="object-item-wrap"
            >
              <div
                className="object-key simple"
              >
                <input
                  className="field key-field"
                  disabled={true}
                  type="text"
                  value="current"
                />
              </div>
              <div
                className="object-value"
              >
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="John Doe"
                  />
                </div>
              </div>
            </div>
            <div
              className="object-item-wrap"
            >
              <div
                className="object-key simple"
              >
                <input
                  className="field key-field"
                  disabled={true}
                  type="text"
                  value="handle"
                />
              </div>
              <div
                className="object-value"
              >
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="JohnDoe"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="object-item-wrap"
      >
        <div
          className="object-key object"
        >
          <input
            className="field key-field"
            disabled={true}
            type="text"
            value="collaborator"
          />
        </div>
        <div
          className="object-value"
        >
          <div
            className="meta-value-object"
          >
            <div
              className="object-item-wrap"
            >
              <div
                className="object-key simple"
              >
                <input
                  className="field key-field"
                  disabled={true}
                  type="text"
                  value="current"
                />
              </div>
              <div
                className="object-value"
              >
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="Jane Doe"
                  />
                </div>
              </div>
            </div>
            <div
              className="object-item-wrap"
            >
              <div
                className="object-key simple"
              >
                <input
                  className="field key-field"
                  disabled={true}
                  type="text"
                  value="handle"
                />
              </div>
              <div
                className="object-value"
              >
                <div
                  className="meta-value"
                >
                  <textarea
                    className="field value-field"
                    disabled={true}
                    onChange={[Function]}
                    style={
                      Object {
                        "height": 0,
                      }
                    }
                    value="JaneDoe"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components::StaticMetaData should render simple field-value correctly 1`] = `
<div
  className="metafields"
  id="defaults"
  title="Default Front Matter"
>
  <div
    className="metafield"
  >
    <div
      className="meta-key simple"
    >
      <input
        className="field key-field"
        disabled={true}
        type="text"
        value="layout"
      />
    </div>
    <div
      className="meta-value"
    >
      <textarea
        className="field value-field"
        disabled={true}
        onChange={[Function]}
        style={
          Object {
            "height": 0,
          }
        }
        value="page"
      />
    </div>
  </div>
</div>
`;
